src/gui_common/dialogs/CustomConfirmationDialog.cs

```diff
@@ -172,6 +175,6 @@ private void OnCancelPressed()
     {
         GUICommon.Instance.PlayButtonPressSound();
         Hide();
-        EmitSignal(nameof(Cancelled));
+        EmitSignal(nameof(Closed));
```

Comment:
Shouldn't "Hide" already trigger the `Closed` signal?

Reply:
No? the signal documentation explicitly states that it's not made for that purpose.

Reply:
That's very confusing to me... `Closed` should *always* trigger after the dialog stops being visible.

Reply:
I think this is a name problem, choosing the name `Canceled` it is...

Reply:
Yeah, the situation will be a lot better if all of the different things have clearly named signals:
- Closed: always when closed
- Cancelled: when rejected
- Accepted or maybe Success

So closed would always be accompanied with one other signal, and the code using the system can then make the decision what it wants to do and when.

Reply:
Done.

Reply:
I feel like the cancelled signal is overused. For example:

```
[connection signal="Canceled" from="PerformanceMetrics" to="." method="OnPerformanceMetricsToggled"]
```
Why does the callback *actually* need to know that the popup was cancelled, rather than it closed for *any* reason? Wouldn't that second one be much, much more bulletproof. The code doesn't care *why* it closed, just that it did and update other GUI state.

Also the signal name should be Cancelled, with two l's as my spell checker complains about "canceled" being a typo.

Reply:
> I feel like the cancelled signal is overused. For example:

I don't know, this was decided by the person who made it. I just tweaked the signal's name.

> Why does the callback actually need to know that the popup was cancelled, rather than it closed for any reason?

So that it's possible to separate out `Confirmed` from the equation. An example of what I'm talking about: you set up a callback registered to a "Closed/Hidden" signal and another callback to a "Confirmed" signal, you clicked confirm and instead of seeing a single operation executed by the "Confirmed" callback, you get both "Confirmed" and "Cancelled" at the same time since "Confirm" also closes the popup most of the time.

Reply:
I agreed about naming it cancelled only for the cancel action. The generic close signal should stay named "Closed". So could you please rename it back to closed and then *only where necessary* would a new "cancelled" signal be used to separate the usage where it is truly needed.

Reply:
Done.

Reply:
Nice, the commit looked fine with a quick glance. I'll fully re-check this PR again soon.